### PR TITLE
DT-1311: 🚨 Fix typecheck warnings

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/stringUtils.js [hunk 1]
```diff
@@ -21,47 +21,36 @@ const toFullName = ({ firstName, lastName, name }) =>
...
-/**
- * Render a moment object or ISO 8601 formatted string in the standard ui date format.  This format is
- * dd/MM/yyyy.  If the date parameter is not present or cannot be understood the function returns '--'.
- * @param date An object which is a moment, an ISO 8601 date formatted string or anything else including null or undefined.
- */
-const renderDate = (date) => {
-  const screenFormat = 'DD/MM/YYYY'
-  const notPresentString = '--'
-
-  const theMoment = typeof date === 'string' ? moment(date, iso8601DateFormat) : date
```
Comment (petergphillips) [inline src/stringUtils.js:33]: the problem here was that `theMoment` wasn't constrained to particular type, could be `any`

File: tsconfig.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "checkJs": true,
     "resolveJsonModule": true,
     "esModuleInterop": true,
-    "allowSyntheticDefaultImports": true
+    "allowSyntheticDefaultImports": true,
+    "jsx": "react"
```
Comment (petergphillips) [inline tsconfig.json:17]: cope with react files

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/stringUtils.js (hunk 1 above)
Comment (andrewrlee) [reply src/stringUtils.js:33]: Might have been a union of `moment.Moment | string`?

You can provide jsdoc type hints if you need to - but the resulting code seems nicer anyway...

**Pull request merged**